### PR TITLE
Add example code for REST API tutorial

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
Related #3420
Let's add a tutorial for REST API!

Modifications:
- Add sample code for REST API tutorial.

Result:
- You can refer the sample code for REST API

### CODE REVIEW

**Review by tuhuynh27 — APPROVED** [review]
LGTM

**Review by ikhoon — APPROVED** [review]
Thanks! @minwoox 
Would you mind resolving the conflicts before merging this PR? 😉

File: examples/tutorials/rest-api-annotated-service/src/main/java/example/armeria/server/blog/BlogPostRequestConverter.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final AtomicInteger idGenerator = new AtomicInteger();
+
+    @Override
+    public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpRequest request,
+                                 Class<?> expectedResultType,
+                                 @Nullable ParameterizedType expectedParameterizedResultType)
+            throws Exception {
+        if (expectedResultType == BlogPost.class) {
```
Comment (trustin) [inline examples/tutorials/rest-api-annotated-service/src/main/java/example/armeria/server/blog/BlogPostRequestConverter.java:30]: How about adding a dedicated class for blog post creation request, so we don't need this converter at all?

File: examples/tutorials/rest-api-annotated-service/build.gradle [hunk 2]
```diff
@@ -0,0 +1,18 @@
+plugins {
+    id 'application'
+}
+
+dependencies {
+    implementation project(':core')
+    implementation 'com.google.guava:guava'
```
[the remark this answers is not in the document]
Comment (minwoox) [reply examples/tutorials/rest-api-annotated-service/build.gradle]: Thanks! Removed it as we discussed. 😄

[the remark this answers is not in the document]
Comment (trustin) [reply examples/tutorials/rest-api-annotated-service/build.gradle]: Thanks!

File: examples/tutorials/rest-api-annotated-service/src/main/java/example/armeria/server/blog/BlogPostRequestConverter.java (hunk 1 above)
Comment (minwoox) [reply examples/tutorials/rest-api-annotated-service/src/main/java/example/armeria/server/blog/BlogPostRequestConverter.java:30]: Let's add another section to the tutorial that we can do this in a simple way as we discussed. 😄

File: examples/tutorials/rest-api-annotated-service/build.gradle [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    runtimeOnly 'org.slf4j:slf4j-simple'
+
+    testImplementation project(':junit5')
+    testImplementation 'net.javacrumbs.json-unit:json-unit-fluent'
+    testImplementation 'org.assertj:assertj-core'
+    testImplementation 'org.junit.jupiter:junit-jupiter-api'
+}
+
+application {
+    mainClassName = 'example.armeria.server.blog.Main'
```
Comment (ikhoon) [inline examples/tutorials/rest-api-annotated-service/build.gradle:16]: `mainClassName` has been deprecated in Gradle 7.x.
```suggestion
    mainClass.set('example.armeria.server.blog.Main')
```

**Review by ikhoon — APPROVED** [review]
Thanks a lot! @minwoox

**Pull request merged**